packages/web/src/pages/edit-page/EditTrackPage.tsx

```diff
@@ -48,8 +48,12 @@ export const EditTrackPage = (props: UploadPageProps) => {
   const [showDeleteConfirmation, setShowDeleteConfirmation] = useState(false)
 
   const onSubmit = (formValues: TrackEditFormValues) => {
-    const metadata = formValues.tracks[0].metadata
+    const metadata = { ...formValues.trackMetadatas[0] }
+    if (!metadata.artwork?.file) {
+      metadata.artwork = null
+    }
     dispatch(editTrack(metadata.track_id, metadata))
```

Comment:
@dharit-tan this is where the submit happens. It was submitting already, but erroring out due to the artwork field being poorly set. Now it's working!